enhancements/storage/storage-hypershift.md

```diff
@@ -0,0 +1,973 @@
+
+### User Stories
+
+* As a guest cluster user, I want the cluster to run as little OCP infrastructure workloads as possible to have the
+  cluster just for my applications.
+  
+* As a guest cluster user, I want the cluster to be as robust as possible, I should not be able to break it
+  by my actions.
+
+* As a guest cluster user, I don't want OCP infrastructure components running there to have elevated RBACs, so
+  a compromised component cannot break my cluster (too much).
+  * Still, CSI driver Pods *must* run on each worker node in the guest cluster as privileged, to be able to format and
+    mount volumes. A compromised CSI driver Pod can do anything to the application data - steal it, delete it, encrypt
+    it..., and it can fully compromise the node.
+    
+* As a HyperShift user, I want to spawn new OCP guest clusters on AWS, so the guest clusters can use AWS EBS storage
+  out of the box.
+  * Explicitly, the guest clusters can run in different AWS regions and zones than the HyperShift management cluster.
+    The management cluster does not need to run on AWS at all.
+
+* As a HyperShift user, I want my OCP guest clusters with no worker nodes so I can 1) autoscale the cluster from zero or
+  2) "hibernate" the cluster when it's idle and resume it later.
+  * Control-plane components should keep working without any issues.
+    * Users should be e.g. able to delete PVCs and expect that the corresponding volume is deleted in AWS.
+    * Dynamic provisioning should work if forced by `volumeBindingMode: Immediate`. Users are then responsible
+      for creating nodes in the right availability zones so that they can use the provisioned volumes.
+
+* Phase 2: As a HyperShift user, I want to get metrics of storage management cluster components so I can monitor them.
+  * This is Phase 2, as most useful storage metrics are already emitted by kube-controller-manager. While metrics of the
+    CSI driver, its sidecars, snapshot-controller and their operators are interesting, they're not crucial.
+
+* Phase 2: As a HyperShift user, I want all management cluster components that talk to a guest cluster to have proper
+  RBACs, so a compromised storage component can't break the guest cluster (too much).
+  * The initial implementation may give cluster-admin role in the guest cluster to storage components in the management
+    cluster - a compromised component in the management cluster could get full access the guest cluster, but not to the
+    management cluster.
+  * Even with the minimal RBACs, CSI driver components will be able to delete any PV and any volume in the cloud,
+    messing up the guest cluster pretty badly.
+
+### Goals
+
+Move storage control plane components to HyperShift management cluster.
+
+### Non-Goals
+
+In Phase 1:
+
+* Metrics of the storage components running in the management cluster.
+* Fine-grained guest cluster RBACs for the storage components running in the management cluster.
```

Comment:
why is this a non goal?

Reply:
It's in Phase2. I want really minimal product in the first release. There is some chance we will have time for RBACs.
A compromised CSI driver component can break the guest cluster pretty badly even if it has minimal RBAC anyway.